### DISCUSSION TITLE
How to compile OpenSSL for Apple VisionOS?

### DISCUSSION THREAD

**Discussion by huakucha** [openssl/openssl#21567] [Q&A] [answered]
VisionOS has been added to ios.toolchain.cmake, but .conf can't work

**Comment by huakucha** [comment, marked as answer]
## Apple visionOS -- visionos -- VISIONOS
    "visionos-cross-base" => {
        inherit_from    => [ "ios-cross-base" ],
        template         => 1,
        defines          => [ "HAVE_FORK=0" ],
    },
    "visionos-sim-cross-x86_64" => {
        inherit_from     => [ "darwin64-x86_64", "visionos-cross-base" ],
        sys_id           => "visionOS",
        perlasm_scheme   => "macosx",
    },
    "visionos-sim-cross-arm64" => {
        inherit_from     => [ "darwin-common", "visionos-cross-base" ],
        cflags           => add("-arch arm64", sub { defined($ENV{'VISIONOS_MIN_SDK_VERSION'}) ? '-target arm64-apple-xros$(VISIONOS_MIN_SDK_VERSION)-simulator' : '-target arm64-apple-xros1.0-simulator'; }),
        sys_id           => "visionOS",
        asm_arch         => 'aarch64',
        perlasm_scheme   => "macosx",
    },

    "visionos-cross-arm64" => {
        inherit_from     => [ "darwin-common", "visionos-cross-base" ],
        cflags           => add("-arch arm64"),
        perlasm_scheme   => "ios64",
        sys_id           => "visionOS",
        asm_arch         => 'aarch64',
    },

**Reply by huakucha** [reply]
I have done visionOS configuration by myself, and it has been able to produce smoothly...

**Reply by xtrachen** [reply]
hello,i use this set ,run make ,has report a file as:
/Applications/Xcode-beta.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/clang/15.0.0/include/inttypes.h:21:15: fatal error: 'inttypes.h' file not found
#include_next <inttypes.h>

I run it in openssl 3.2.0 & Xcode 15.1 beta3,is where need update?

**Comment by serajahmad01** [comment]
Hi @huakucha 
Can you help me as well in this https://github.com/openssl/openssl/issues/22110

Thanks in advance
